lib/listen/silencer.rb

```diff
@@ -34,6 +34,9 @@ class Silencer
       | \.swpx
       | ^4913
 
+      # Emacs backup/swap files
+      | (?:\.\#.+|\#.+\#)
```

Comment:
I believe these should be anchored per our discussion on the Issue. And \A and \z are the best way to anchor in Ruby so as not to be tripped up by newlines (which are rare in filenames...but possible). So I think this does it:
```
(?:\A\.\#.|\A\#.+\#\z)
```

Reply:
@ColinDKelley , oh, i add to wrong place, i thought where i insert in regexp is around with a `^...$`

for \A, \z, agree with you, but why i can't find those anchor within exists regexp? (only see some `^`, `$` anchor out of group)

Reply:
I'm guessing that those who wrote the earlier code didn't know about the Ruby trap of `^` and `$` not working the same as in every other language.

I can also believe the some of the existing patterns weren't anchored at all, even when they should have been.

Reply:
okay, so, double confirm, you want me fix those exists regexp anchor, and then add emacs pattern into it, right?

Reply:
Yes, two separate commits should make it clear:

1. Fix the existing ^ and $ anchors to \A and \z.
2. Add your new emacs patterns with \A and \z anchors.

Or if you'd rather keep this PR simple, we can flip the order and I can take care of (2):

1. You add your new emacs patterns with ^ and $ anchors.
2. [after merging this PR] I fix all the ^ and $ anchors to \A and \z.

Reply:
Got it, let me do both of them first anyway.